src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final float CAPACITY_GROWTH = 1.5f;
+    private static final ArrayListIndexOutOfBoundsException
+            ARRAY_LIST_INDEX_OUT_OF_BOUNDS_EXCEPTION
+            = new ArrayListIndexOutOfBoundsException(
+                    "Given index is not valid for size");
+    private T[] elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (size >= elementData.length) {
+            grow();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Given index is not valid for size: " + size);
+        }
+        size++;
+        if (size == elementData.length) {
+            grow();
+        }
+        T[] oldElementData = elementData;
+        if (size - (index + 1) >= 0) {
+            System.arraycopy(oldElementData,
+                    index + 1 - 1,
+                    elementData,
+                    index + 1,
+                    size - (index + 1));
+        }
```

Comment:
please, simplify this piece of code. Also you don't need to have an additional array for System.arraycopy call